free-2.md

```diff
@@ -1,69 +1,168 @@
 # Free project 2
 
 
+
+This project would be something like the functional equivalent of [scratch][scratch.mit.edu], only as a learning tool for textual coding. It would have a split-screen interface with the area for writing code on the left (maybe in a new language, but most likely in a pre-existing one), and on the right would be an image visualiation of the code, with similar color-coded "building blocks" to show exactly what functions are being called where and what's being passed into them. As code is added, there could be a button to "refresh" the visualization area, which would also be updated when the code is executed.
```

Comment:
This sounds like a very exciting idea. I feel very comfortable with functional programming, but also do have a very hard time tying the essence of it to any visual representation.

This would probably be a unique and challenging project!